### PR TITLE
feat: block for deployment queue processing

### PR DESCRIPTION
**Pull request by danielpeach**
## How has this been tested? How can a reviewer test these changes?

# Have your performed the following tests?

Please confirm you have done the following tests to ensure your pull request is ready to be merged..

- [ ] If this change modifies the deployment flow have triggered a deployment using this branch (i.e. `build/bin/darwin_amd64/armory deploy start -f <path to file>`)?
- [ ] If this change modifes the login flow have you tried logging in and out with this branch (i.e. `build/bin/darwin_amd64/armory login`) and verified the credentials work?
- [ ] Have you verified the specific change made in this PR?

### CODE REVIEW

File: cmd/deploy/start.go [hunk 1]
```diff
@@ -119,6 +131,13 @@ func NewDeployStartCmd(configuration *config.Configuration) *cobra.Command {
 			cmdUtils.ExecuteParentHooks(cmd, args)
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if !*configuration.GetIsTest() {
```
Comment (danielpeach) [inline cmd/deploy/start.go:134]: This is horribly ugly. The cli needs a big refactor.

File: cmd/deploy/waiter.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	cloudConsoleBaseURL string,
+) *Waiter {
+	return &Waiter{
+		client:              gqlclient,
+		tick:                defaultWaiterTickDuration,
+		cloudConsoleBaseURL: cloudConsoleBaseURL,
+	}
+}
+
+func (w *Waiter) WaitForPipelineToBeProcessed(ctx context.Context, pipelineID string) error {
```
Comment (Greg-At-Armory) [inline cmd/deploy/waiter.go:44]: why bother with this custom implementation and not use stuff from our go-commons awaitability package?

**Review by Greg-At-Armory — APPROVED** [review]
except potentially not needed waiter.go i don't see any red flags.

File: internal/graphql/pipelines.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	  id
+	  status
+      blockedByPipeline {
+		id
+      }
+      application {
+	    name
+      }
+    }
+  }
```
Comment (aelath) [inline internal/graphql/pipelines.go]: minor: looks like the indentation is weird here.

**Review by aelath — APPROVED** [review]
I was able to verify that this works as expected. I could both start & replace pipelines, as well as get a nice error message when strategy was changed to `reject`. 🌮

**Pull request merged**